element_interface/prairieviewreader.py

```diff
@@ -0,0 +1,159 @@
+
+    The PrairieView software generates one .ome.tif imaging file per frame acquired. The metadata for all frames is contained one .xml file. This function locates the .xml file and generates a dictionary necessary to populate the DataJoint ScanInfo and Field tables.
+
+    PrairieView works with resonance scanners with a single field.
+
+    PrairieView does not support bidirectional x and y scanning.
+
+    ROI information is not contained in the .xml file.
+
+    All images generated using PrairieView have square dimensions (e.g. 512x512).
+
+
+    Args:
+        pvtiffile: An absolute path to the .ome.tif image file.
+
+    Raises:
+        FileNotFoundError: No .xml file containing information about the acquired scan was found at path in parent directory at `pvtiffile`.
+
+    Returns:
+        metainfo: A dict mapping keys to corresponding metadata values fetched from the .xml file.
+    """
+
+    xml_files = pathlib.Path(pvtiffile).parent.glob("*.xml") # May return multiple xml files. Only need one that contains scan metadata.
+
+    for xml_file in xml_files :
+        tree = ET.parse(xml_file)
+        root = tree.getroot()
+        if root.find(".//Sequence"):
+            break
+    else:
+        raise FileNotFoundError(
+            f"No PrarieView metadata XML file found at {pvtiffile.parent}"
+        )
+
+    bidirectional_scan = False  # Does not support bidirectional
+
+    nfields = 1  # Always contains 1 field
+
+    # Get all channels and find unique values
+    channel_list = []
+    channels = root.iterfind(".//Sequence/Frame/File/[@channel]")
+    for channel in channels:
+        channel_list.append(int(channel.attrib.get("channel")))
+    nchannels = np.unique(channel_list).shape[0]
+
+    # One "Frame" per depth. Gets number of frames in first sequence
+    planes_list = []
+    planes = root.findall(".//Sequence/[@cycle='1']/Frame")
+    for plane in planes:
+        planes_list.append(int(plane.attrib.get("index")))
+    ndepths = np.unique(planes_list).shape[0]
+
+    # Total frames are displayed as number of "cycles"
+    nframes = len(root.findall(".//Sequence/Frame"))
+
+    roi = 1
+
+    x_coordinate = float(
+        root.find(
+            ".//PVStateValue/[@key='currentScanCenter']/IndexedValue/[@index='XAxis']"
+        ).attrib.get("value")
+    )
+    y_coordinate = float(
+        root.find(
+            ".//PVStateValue/[@key='currentScanCenter']/IndexedValue/[@index='YAxis']"
+        ).attrib.get("value")
+    )
+    z_coordinate = float(
+        root.find(
+            ".//PVStateValue/[@key='positionCurrent']/SubindexedValues/[@index='ZAxis']/SubindexedValue/[@subindex='0']"
+        ).attrib.get("value")
+    )
+
+    framerate = np.divide(
+        1,
+        float(
+            root.findall('.//PVStateValue/[@key="framePeriod"]')[0].attrib.get("value")
+        ),
+    )  # rate = 1/framePeriod
```

Comment:
What's the reason for `np.divide`?
```suggestion
    framerate = 1 / float(
            root.findall('.//PVStateValue/[@key="framePeriod"]')[0].attrib.get("value")))  # rate = 1/framePeriod
```

Reply:
I'm very used to using `numpy` for my math operations. We can do `1/period` and get the same result. The file just doesn't provide an outright frame rate and instead makes us calculate it